pincer/objects/guild/guild.py

```diff
@@ -1962,6 +1964,123 @@ async def get_webhooks(self) -> AsyncGenerator[Webhook, None]:
+        for event_data in data:
+            yield ScheduledEvent.from_dict(
+                construct_client_dict(self._client, event_data)
+            )
+
+    async def create_scheduled_event(
+        self,
+        name: str,
+        privacy_level: int,
+        entity_type: int,
+        scheduled_start_time: datetime,
+        scheduled_end_time: Optional[datetime] = None,
+        entity_metadata: Optional[str] = None,
+        channel_id: Optional[Snowflake] = None,
+        description: Optional[str] = None,
+        reason: Optional[str] = None,
+    ) -> ScheduledEvent:
+        if scheduled_start_time.timestamp() < datetime.now().timestamp():
+            raise ValueError("An event cannot be created in the past")
+
+        if scheduled_end_time is not None and scheduled_end_time.timestamp() < datetime.now().timestamp():
+            raise ValueError("An event cannot start before it ends")
+
+        data = await self._http.post(
+            f"guilds/{self.id}/scheduled-events",
+            data={
+                "name": name,
+                "privacy_level": privacy_level,
+                "entity_type": entity_type,
+                "scheduled_start_time": scheduled_start_time.isoformat(),
+                "scheduled_end_time": scheduled_end_time.isoformat() if scheduled_end_time is not None else None,
+                "entity_metadata": entity_metadata,
+                "channel_id": channel_id,
+                "description": description,
+                },
+            headers={"X-Audit-Log-Reason": reason},
+            )
+        return ScheduledEvent.from_dict(
+            construct_client_dict(self._client, data)
+        )
+
+    async def get_scheduled_event(self, _id: Snowflake, with_user_count: bool = False) -> ScheduledEvent:
+        data = await self._http.get(f"guilds/{self.id}/scheduled-events/{_id}?{with_user_count=!s}")
+        return ScheduledEvent.from_dict(
+            construct_client_dict(self._client, data)
+        )
+
+    async def modify_scheduled_event(
+        self,
+        _id: Snowflake,
+        name: Optional[str] = None,
+        entity_type: Optional[int] = None,
+        privacy_level: Optional[int] = None,
+        scheduled_start_time: Optional[datetime] = None,
+        scheduled_end_time: Optional[datetime] = None,
+        entity_metadata: Optional[str] = None,
+        channel_id: Optional[Snowflake] = None,
+        description: Optional[str] = None,
+        status: Optional[int] = None,
+        reason: Optional[str] = None,
+    ) -> ScheduledEvent:
+        if scheduled_start_time is not None and scheduled_start_time.timestamp() < datetime.now().timestamp():
+            raise ValueError("An event cannot be created in the past")
+
+        if scheduled_end_time is not None and scheduled_end_time.timestamp() < datetime.now().timestamp():
+            raise ValueError("An event cannot start before it ends")
```

Comment:
```suggestion
        if scheduled_start_time is not None:
            timestamp = datetime.now().timestamp()
            if scheduled_start_time.timestamp() < timestamp:
                raise ValueError("An event cannot be created in the past")

            if scheduled_end_time.timestamp() < timestamp:
                raise ValueError("An event cannot start before it ends")
```
Avoiding calculating twice the same thing